server/szurubooru/func/pools.py

```diff
@@ -0,0 +1,318 @@
+        return self.pool.last_edit_time
+
+    def serialize_post_count(self) -> Any:
+        return self.pool.post_count
+
+    def serialize_posts(self) -> Any:
+        return [post for post in
+                [posts.serialize_micro_post(rel, None)
+                 for rel in self.pool.posts]]
+
+
+def serialize_pool(
+        pool: model.Pool, options: List[str] = []) -> Optional[rest.Response]:
+    if not pool:
+        return None
+    return PoolSerializer(pool).serialize(options)
+
+
+def try_get_pool_by_id(pool_id: int) -> Optional[model.Pool]:
+    return (
+        db.session
+        .query(model.Pool)
+        .filter(model.Pool.pool_id == pool_id)
+        .one_or_none())
+
+
+def get_pool_by_id(pool_id: int) -> model.Pool:
+    pool = try_get_pool_by_id(pool_id)
+    if not pool:
+        raise PoolNotFoundError('Pool %r not found.' % pool_id)
+    return pool
+
+
+def try_get_pool_by_name(name: str) -> Optional[model.Pool]:
+    return (
+        db.session
+        .query(model.Pool)
+        .join(model.PoolName)
+        .filter(sa.func.lower(model.PoolName.name) == name.lower())
+        .one_or_none())
+
+
+def get_pool_by_name(name: str) -> model.Pool:
+    pool = try_get_pool_by_name(name)
+    if not pool:
+        raise PoolNotFoundError('Pool %r not found.' % name)
+    return pool
+
+
+def get_pools_by_names(names: List[str]) -> List[model.Pool]:
+    names = util.icase_unique(names)
+    if len(names) == 0:
+        return []
+    return (
+        db.session.query(model.Pool)
+        .join(model.PoolName)
+        .filter(
+            sa.sql.or_(
+                sa.func.lower(model.PoolName.name) == name.lower()
+                for name in names))
+        .all())
+
+
+def get_or_create_pools_by_names(
+        names: List[str]) -> Tuple[List[model.Pool], List[model.Pool]]:
+    names = util.icase_unique(names)
+    existing_pools = get_pools_by_names(names)
+    new_pools = []
+    pool_category_name = pool_categories.get_default_category_name()
+    for name in names:
+        found = False
+        for existing_pool in existing_pools:
+            if _check_name_intersection(
+                    _get_names(existing_pool), [name], False):
+                found = True
+                break
+        if not found:
+            new_pool = create_pool(
+                names=[name],
+                category_name=pool_category_name,
+                post_ids=[])
+            db.session.add(new_pool)
+            new_pools.append(new_pool)
+    return existing_pools, new_pools
+
+
+def delete(source_pool: model.Pool) -> None:
+    assert source_pool
+    db.session.delete(source_pool)
+
+
+def merge_pools(source_pool: model.Pool, target_pool: model.Pool) -> None:
+    assert source_pool
+    assert target_pool
+    if source_pool.pool_id == target_pool.pool_id:
+        raise InvalidPoolRelationError('Cannot merge pool with itself.')
+
+    def merge_posts(source_pool_id: int, target_pool_id: int) -> None:
```

Comment:
Better to name `merge_pool_posts` as this is a dangerous name clash with another function